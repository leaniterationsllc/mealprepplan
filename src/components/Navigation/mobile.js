/**
 * Copyright (c) 2017-present, Lean Iterations LLC.
 * All rights reserved.
 *
 * This source code is licensed under the MIT-style license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

class MobileNav extends Component {
  constructor(props) {
    super(props);
    this.mobileContainer = document.createElement('div');
    document.body.appendChild(this.mobileContainer);
  }

  componentWillUnmount() {
    document.body.removeChild(this.mobileContainer);
  }

  render() {
    return ReactDOM.createPortal(
      <nav className={this.props.mobileStyle}>
        <div
          onClick={this.props.toggleNav}
          onKeyPress={this.props.toggleNav}
          role="button"
          tabIndex="0"
        >
          <i className="fas fa-times-circle fa-lg" />
        </div>
        {this.props.children}
      </nav>,
      this.mobileContainer
    );
  }
}

MobileNav.propTypes = {
  mobileStyle: PropTypes.string.isRequired,
  toggleNav: PropTypes.func.isRequired,
  children: PropTypes.element.isRequired
};

export default MobileNav;
