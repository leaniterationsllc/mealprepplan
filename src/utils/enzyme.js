/**
 * Copyright (c) 2017-present, Lean Iterations LLC.
 * All rights reserved.
 *
 * This source code is licensed under the MIT-style license found in the
 * LICENSE file in the root directory of this source tree.
 */
// configuring enzyme to use the adapter for React 16
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

export default Enzyme.configure({ adapter: new Adapter() });
