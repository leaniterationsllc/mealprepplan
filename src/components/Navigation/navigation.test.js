import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navigation from './';
import enzymeconf from '../../utils/enzyme';
import MobileNav from './mobile';

describe('Navigation', () => {
  it('renders', () => {
    const wrapper = shallow(<Navigation />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the mobile nav', () => {
    const wrapper = shallow(<Navigation />);
    wrapper.setState({ mobileActive: true });

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('toggles the mobile nav - show', () => {
    const wrapper = shallow(<Navigation />);

    wrapper.find('div[role="button"]').simulate('click');
    expect(wrapper.state('mobileActive')).toBeTruthy();
  });

  it('renders the mobile nav - hide', () => {
    const wrapper = shallow(<Navigation />);
    wrapper.setState({ mobileActive: true });

    wrapper.find(MobileNav).prop('toggleNav')();
    expect(wrapper.state('mobileActive')).toBeFalsy();
  });
});
