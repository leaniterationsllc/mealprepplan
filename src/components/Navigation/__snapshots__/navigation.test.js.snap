// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders 1`] = `
<nav>
  <Styled(div)
    className="css-eeuixl"
    px={
      Array [
        3,
        3,
        4,
      ]
    }
    width="100%"
  >
    <ul>
      <li>
        Meal Prep Planner and Diary
      </li>
      <div>
        <li>
          <GatsbyLink
            to="/"
          >
            Home
          </GatsbyLink>
        </li>
        <li>
          <GatsbyLink
            to="/store"
          >
            Store
          </GatsbyLink>
        </li>
      </div>
    </ul>
  </Styled(div)>
  <Styled(div)
    className="css-3brnur"
    px={
      Array [
        3,
        3,
        4,
      ]
    }
    width="100%"
  >
    <ul>
      <li>
        Meal Prep Planner and Diary
      </li>
      <li>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <i
            className="fas fa-bars"
          />
        </div>
      </li>
    </ul>
  </Styled(div)>
</nav>
`;

exports[`Navigation renders the mobile nav 1`] = `
<nav>
  <Styled(div)
    className="css-eeuixl"
    px={
      Array [
        3,
        3,
        4,
      ]
    }
    width="100%"
  >
    <ul>
      <li>
        Meal Prep Planner and Diary
      </li>
      <div>
        <li>
          <GatsbyLink
            to="/"
          >
            Home
          </GatsbyLink>
        </li>
        <li>
          <GatsbyLink
            to="/store"
          >
            Store
          </GatsbyLink>
        </li>
      </div>
    </ul>
  </Styled(div)>
  <Styled(div)
    className="css-3brnur"
    px={
      Array [
        3,
        3,
        4,
      ]
    }
    width="100%"
  >
    <ul>
      <li>
        Meal Prep Planner and Diary
      </li>
      <li>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <i
            className="fas fa-bars"
          />
        </div>
      </li>
    </ul>
  </Styled(div)>
  <MobileNav
    mobileStyle="css-gkliy9"
    toggleNav={[Function]}
  >
    <ul>
      <li>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <GatsbyLink
            to="/"
          >
            Home
          </GatsbyLink>
        </div>
      </li>
      <li>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="-1"
        >
          <GatsbyLink
            to="/store"
          >
            Store
          </GatsbyLink>
        </div>
      </li>
    </ul>
  </MobileNav>
</nav>
`;
